<launch>
    <!-- Whether to load the URDF -->
    <arg name="load_robot_description" default="false"/>

    <!-- Which robot the simulation is for -->
    <arg name="robot" default="roman"/>

    <!-- Optionally, load the URDF for the default RoMan (roman1) -->
    <include if="$(arg load_robot_description)" file="$(find roman_model)/robots/upload_roman1.launch"/>

    <!-- Spawn the main simulation node that provides a surrogate, kinematic simulation of the robot -->
    <node name="robot_simulator" pkg="roman_simulation" type="robot_simulator"/>

    <!-- Spawn controllers for the specific robot -->
    <include file="$(find roman_simulation)/launch/robot_simulator_$(arg robot).launch"/>

    <!-- Publish /tf for internal joint transforms -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    <!-- Publish a static transform for localization (odometry provided by controllers) -->
    <node name="fake_localization" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 map odom 16"/>

    <!-- Spawn trajectory controller wrapper node that accepts under-specified trajectories -->
    <node name="trajectory_dispatcher" pkg="roman_simulation" type="trajectory_dispatcher">
        <rosparam command="load" file="$(find roman_simulation)/config/trajectory_dispatcher_$(arg robot).yaml"/>
    </node>

    <!-- Spawn an action server for simple gripper commands -->
    <node name="gripper_command_action_server" pkg="roman_simulation" type="gripper_command_action_server"/>
</launch>
